lib/index.d.ts

```diff
@@ -0,0 +1,228 @@
+
+  constructor(width?: number, height?: number)
+
+  /**
+   * Cast this object as a {@link SyncCanvas} and set the `async` property to false to get the synchronous behaviours.
+   *
+   * @example
+   * import { Canvas, SyncCanvas } from "."
+   * const myCanvas = new Canvas() as any as SyncCanvas
+   * myCanvas.async = false
+   * const result = myCanvas.toBuffer("png") // now these functions return synchronously
+   */
+  async: true
+
+  get width(): number
+  get height(): number
+
+  getContext(type?: "2d"): CanvasRenderingContext2D
+
+  get pages(): readonly CanvasRenderingContext2D[]
+
+  get pdf(): Promise<Buffer>
+  get svg(): Promise<Buffer>
+  get jpg(): Promise<Buffer>
+  get png(): Promise<Buffer>
+
+  newPage(width?: number, height?: number): CanvasRenderingContext2D
+
+  saveAs(filename: string, options?: SaveOptions): Promise<void>
+  toBuffer(format: string, options?: RenderOptions): Promise<Buffer>
+  toDataURL(format: string, options?: RenderOptions): Promise<string>
+}
+
+export type SyncCanvas = {
+  [P in keyof Canvas]: Canvas[P] extends Promise<infer Value>
+    ? Value // Promise getter to synchronous getter
+    : Canvas[P] extends (...args: infer Args) => Promise<infer Return>
+    ? (...args: Args) => Return // Async functions to sync functions
+    : P extends "async"
+    ? false // `async` property is now false
+    : Canvas[P] // Everything else stays the same
+}
+
+export interface CreateTextureOptions {
+  path?: Path2D
+  line?: number
+  color?: string
+  angle?: number
+  offset?: [x: number, y: number]
+}
+
+export class CanvasRenderingContext2D extends globalThis.CanvasRenderingContext2D {
+  // @ts-expect-error We're rewriting the canvas property in a non-typesafe way
+  readonly canvas: Canvas
+
+  fontVariant: string
+  textTracking: number
+  textWrap: boolean
+
+  lineDashFit: "move" | "turn" | "follow"
+  lineDashMarker: Path2D
+
+  conicCurveTo(
+    cpx: number,
+    cpy: number,
+    x: number,
+    y: number,
+    weight: number
+  ): void
+
+  createTexture(
+    spacing: number | [width: number, height: number],
+    options?: CreateTextureOptions
+  ): CanvasTexture
+
+  measureText(text: string, maxWidth?: number): TextMetrics
+
+  outlineText(text: string): Path2D
+
+  // @ts-expect-error We're rewriting the canvas property in a non-typesafe way
+  strokeStyle:
+    | globalThis.CanvasRenderingContext2D["strokeStyle"]
+    | CanvasTexture
+  // @ts-expect-error We're rewriting the canvas property in a non-typesafe way
+  fillStyle: globalThis.CanvasRenderingContext2D["fillStyle"] | CanvasTexture
+
+  /** @internal */
+  lineStyle: string
+}
+
+export interface TextMetrics extends globalThis.TextMetrics {
+  lines: TextMetricsLine[]
+}
+
+export interface TextMetricsLine {
+  readonly x: number
+  readonly y: number
+  readonly width: number
+  readonly height: number
+  readonly baseline: number
+  readonly startIndex: number
+  readonly endIndex: number
+}
+
+export class CanvasTexture {}
```

Comment:
This should have some kind of description. Otherwise TypeScript will just accept an empty object `{}` as a texture.